dpcpp/solver/cb_gmres_kernels.dp.cpp

```diff
@@ -453,7 +457,9 @@ void multinorm2_inf_kernel(
             reduction_helper_array_acc_ct1(cgh);
 
         cgh.parallel_for(
-            sycl_nd_range(grid, block), [=](sycl::nd_item<3> item_ct1) {
+            sycl_nd_range(grid, block), [=
+        ](sycl::nd_item<3> item_ct1) [[sycl::reqd_sub_group_size(
+                                            default_dot_dim)]] {
```

Comment:
Shouldn't we use `config::warp_size` here? It's the same size for now. Are there performance issues for it to be exactly 16?
```suggestion
                                            config::warp_size)]] {
```

Reply:
The correctness of these kernels internally depends on the value of default_dot_dim, so I don't want to touch this right now. But in general, I was planning to move all the reductions in #861 anyways, so this code may not be that relevant in the long run?